internal/venice-common/src/main/java/com/linkedin/venice/pushmonitor/PushStatusDecider.java

```diff
@@ -235,6 +237,16 @@ protected ExecutionStatus getPartitionStatus(
     return STARTED;
   }
 
+  boolean isPushjobKilled(List<StatusSnapshot> snapshotList) {
+    for (StatusSnapshot snapshot: snapshotList) {
+      if (snapshot.getStatus() == ERROR
+          && snapshot.getIncrementalPushVersion().contains("signal to kill this consumer")) {
```

Comment:
I guess string comparison is the best we can do ATM, given the way these components communicate, but hardcoding the string is too brittle IMHO. I suggest we create a constant containing `Received the signal to kill this consumer.` and have both places in the code refer to that. We should also add JavaDoc to that constant saying that this string is sent by the server and received by the controller, and should therefore not be changed otherwise the controller will fail to pick up the signal properly if server and controller are running different versions of the code...

Reply:
I was just thinking about it, many codebases have centralized Log messages class with some log-ID for each log and rest of the code uses that. Which makes things like canary tests/log stats deliver a better picture of the current state. Also log rate limiting also gets cleaner as it will try to limit by log-id, probably can be universally rate limited for all logs. Anyways will update the code.